qiskit_experiments/base_analysis.py

```diff
@@ -16,25 +16,40 @@
 from abc import ABC, abstractmethod
 from typing import List, Tuple
 
+from qiskit.providers.options import Options
 from qiskit.exceptions import QiskitError
 
-from .experiment_data import ExperimentData, AnalysisResult
+from qiskit_experiments.experiment_data import ExperimentData, AnalysisResult
+
+# pylint: disable = unused-import
+from qiskit_experiments.matplotlib import pyplot
 
 
 class BaseAnalysis(ABC):
-    """Base Analysis class for analyzing Experiment data."""
+    """Base Analysis class for analyzing Experiment data.
+
+    When designing Analysis subclasses default values for any kwarg
+    analysis options of the `run` method should be set by overriding
+    the `_default_options` class method. When calling `run` these
+    default values will combined with all other option kwargs in the
```

Comment:
```suggestion
    default values will be combined with all other option kwargs in the
```

Reply:
Perhaps for another PR but this docstring could do with an extra few sentences on what the `BaseAnaysis` class is and what it does. E.g. something like:

The data produced by experiments (i.e. subclasses of BaseExperiment) are analyzed with subclasses of
BaseExperiment. The analysis is typically run after the data has been gathered by the experiment. For example,
an analysis may perform some data processing of the measured data and a fit to a function to extract a parameter.